content/learn/05.communication/07.modbus/modbus.md

```diff
@@ -0,0 +1,124 @@
+
+You have to check the device-specific documentation to attain the right address, function code, starting registers and register count. The CRC error check is taken care of by the Modbus library. To make it easier to understand the example below shows how to use the Modbus library.
+
+### Example
+Let's say you have a Modbus-compatible energy meter working with RS-485. In our case, we use a model from [Finder](https://cdn.findernet.com/app/uploads/2021/09/20090052/Modbus-7M24-7M38_v2_30062021.pdf) which uses Modbus RTU protocol but you can use a different one. 
+
+Take a look at the request frame inside the Finder documentation:
+
+![](assets/finderRequestFrame.png)
+
+We can see that the device address is 33 (or 0x21).
+
+The function code states 04 which we know that it stands for INPUT REGISTERS ([see Function Code](#function-code)).
+
+The starting register is 00 6B (or 0x6B) but for the Modbus library to work we need to change it from hexadecimal to decimal.
+In this example, the starting register is stated as hexadecimal but if we scroll down further in the documentation we can see that the registers are written in decimals: 
+
+![](assets/modbusMeasurmentFinder.png)
+
+We can also see that the register for U1 takes up 30107 and 30108 which means that the register count is 2.
+
+Using the `readVoltage()` function, defined within the parameters we discussed, will retrieve the voltage measurement from the Finder energy meter. 
+
+Inside you can see how the `requestFrom()` is being called with *Device Address*, *Function Code*, *Starting Register* and *Register Count* as parameters.
+
+```
+float readVoltage() {
+  float volt = 0.;
+  // Send reading request over RS485 
+  if (!ModbusRTUClient.requestFrom(0x21, INPUT_REGISTERS, 30017, 2)) {
+    // Error handling
+    Serial.print("- Failed to read the voltage! ");
+    Serial.println(ModbusRTUClient.lastError()); 
+  } else {
+    // Response handler 
+    uint16_t word1 = ModbusRTUClient.read();  // Read word1 from buffer
+    uint16_t word2 = ModbusRTUClient.read();  // Read word2 from buffer
+    uint32_t millivolt = word1 << 16 | word2; // Join word1 and word2 to retrieve voltage value in millivolts
+    volt = millivolt/1000.0;                  // Convert to volts
+  }
+
+  return volt;
+}
+```
+
+The following section gives an overview of Modbus compatible Arduino boards and the libraries that you can use to enable Modbus protocol capability. Depending on the hardware you are using, the libraries might vary, therefore it is important to always check your device specifications. As this article does not cover all types of Modbus implementation and mostly focuses on Modbus RTU, the following section includes more references to read about the different types of implementation in more detail.
+
+### Modbus compatible Boards
```

Comment:
I would move this section to the top, after "introduction" (see my other comment).

```suggestion
### Modbus Compatible Hardware
```